### PR TITLE
PLT-4524 Update password reset page

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
This PR implement the Jira ticket PLT-4524, which redirect the user when request to reset the password but their email doesn't exist in mattermost database

#### Ticket Link
Jira - https://mattermost.atlassian.net/browse/PLT-4524
GitHub - https://github.com/mattermost/platform/issues/4497

#### Checklist
- [X] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [ ] Has UI changes
- [X] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by cpanato** [comment]
I also tested on Mac Sierra / Chrome

**Comment by cpanato** [comment]
cool @jasonblais 
:D

File: webapp/components/password_reset_send_link.jsx [hunk 1]
```diff
@@ -45,25 +45,46 @@ class PasswordResetSendLink extends React.Component {
...
-                        </div>
-                    )
-                });
+            (response) => {
+                if (response.email === '') {
+                    this.setState({
+                        error: null,
+                        updateText: (
+                            <div className='reset-form alert alert-success'>
+                                <FormattedHTMLMessage
```
Comment (enahum) [inline webapp/components/password_reset_send_link.jsx]: can you please move out the `<p>` tags from here?

File: webapp/components/password_reset_send_link.jsx [hunk 2]
```diff
@@ -45,25 +45,46 @@ class PasswordResetSendLink extends React.Component {
...
+                                    id='password_send.invalidemail'
+                                    defaultMessage='<p>If the account exists, a password reset email will be sent to <b>{email}</b></p>'
+                                    values={{
+                                        email
+                                    }}
+                                />
+                                <FormattedMessage
+                                    id={'password_send.checkInbox'}
```
Comment (enahum) [inline webapp/components/password_reset_send_link.jsx]: here the `id` and `defaultMessage` should be without the curly braces :)

File: webapp/components/password_reset_send_link.jsx [hunk 3]
```diff
@@ -45,25 +45,46 @@ class PasswordResetSendLink extends React.Component {
...
+                        )
+                    });
+                } else {
+                    this.setState({
+                        error: null,
+                        updateText: (
+                            <div className='reset-form alert alert-success'>
+                                <FormattedHTMLMessage
+                                    id='password_send.link'
+                                    defaultMessage='<p>A password reset link has been sent to <b>{email}</b></p>'
```
Comment (enahum) [inline webapp/components/password_reset_send_link.jsx]: same here

File: webapp/components/password_reset_send_link.jsx [hunk 4]
```diff
@@ -45,25 +45,46 @@ class PasswordResetSendLink extends React.Component {
...
+                                    values={{
+                                        email
+                                    }}
+                                />
+                                <FormattedMessage
+                                    id={'password_send.checkInbox'}
+                                    defaultMessage={'Please check your inbox.'}
```
Comment (enahum) [inline webapp/components/password_reset_send_link.jsx]: and here

File: webapp/i18n/en.json [hunk 5]
```diff
@@ -1554,6 +1554,7 @@
   "password_send.email": "Email",
   "password_send.error": "Please enter a valid email address.",
   "password_send.link": "<p>A password reset link has been sent to <b>{email}</b></p>",
+  "password_send.invalidemail": "<p>If the account exists, a password reset email will be sent to <b>{email}</b></p>",
```
Comment (enahum) [inline webapp/i18n/en.json]: don't forget to match this without the `<p>` tag

File: webapp/components/password_reset_send_link.jsx (hunk 1 above)
Comment (cpanato) [inline webapp/components/password_reset_send_link.jsx]: removed the <p>

File: webapp/components/password_reset_send_link.jsx (hunk 2 above)
Comment (cpanato) [inline webapp/components/password_reset_send_link.jsx]: done

File: webapp/i18n/en.json (hunk 5 above)
Comment (cpanato) [inline webapp/i18n/en.json]: removed and remove the line above as well

**Comment by cpanato** [comment]
Np,  i also agree :)

**Comment by jasonblais** [comment]
Hey @cpanato,

The intention was to update the reset password message to be `If the account exists, a password reset email will be sent to: ...` whether or not an email was sent.

Really sorry as this might have been unclear on the original issue specification.

I've included a clearer spec below, would you be able help with an update?

----

If I
1. Click "I forgot my password"
2. Enter an email (whether or not if has or doesn't have an account on the server)

the user is redirected to the "reset password" page with text

```
If the account exists, a password reset email will be sent to: 
[email address]

Please check your inbox.
```

Below is a screenshot of what the final message on the "reset password" page would look like

image


----

Moving back to PM review for now

**Comment by cpanato** [comment]
ahh! got it, will update the code :)

**Comment by jasonblais** [comment]
Tested on Windows 10, Chrome:
 - user redirected to password reset page whether an account for that email exists or not
 - password reset also works correctly

Minor note : 

1) Click "I forgot my password", then enter an email

Observed: On a large window, the very last word in `If the account exists, a password reset email will be sent to: ` wraps on a second line, which is unnecessary. 

This doesn't reproduce on the `master` branch since the existing text is slightly shorter and fits within the box.

image

Wondering if @cpanato you'd like to take a look and increase the width of the green box just a bit so that the first sentence fits on one line?

I can also ask our UX designer help tweak it.

Let me know which option you'd prefer, both are totally fine.

**Comment by cpanato** [comment]
@jasonblais I did the change, please check if this is good, I'm not an expert on css :/ if not can you please ask the UX?
thanks

**Comment by asaadmahmood** [comment]
@cpanato You can just increase the `max-width` to `400px` of `.signup-team__container`, rather than increasing the width of the alert box. That would fix the smaller screen issue.

**Comment by cpanato** [comment]
@asaadmahmood I didnt know that. Will do that
thanks!

**Pull request merged**